distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json/Extensions.enso

```diff
@@ -1,136 +1,137 @@
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Number.to_js_object : JS_Object | Number
+Number.to_js_object self = case self of
+    Number -> JS_Object.from_pairs [["type", "Number"]]
+    Integer -> JS_Object.from_pairs [["type", "Integer"]]
+    Decimal -> JS_Object.from_pairs [["type", "Decimal"]]
+    _ -> self
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Boolean.to_js_object : JS_Object | Boolean
+Boolean.to_js_object self = case self of
+    Boolean -> JS_Object.from_pairs [["type", "Boolean"]]
+    _ -> self
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Nothing.to_js_object : Nothing
+Nothing.to_js_object self = self
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms and maps, the object is converted to a JS_Object.
+Array.to_js_object : Vector
+Array.to_js_object self =
+    proxy = Array_Proxy.new self.length i-> self.at i . to_js_object
+    Vector.from_polyglot_array proxy
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms, the object is converted to a JS_Object.
+Vector.to_js_object : Vector
+Vector.to_js_object self =
+    proxy = Array_Proxy.new self.length i-> self.at i . to_js_object
+    Vector.from_polyglot_array proxy
+
+## Converts the given value to a JSON serializable object.
+   For Nothing, booleans, numbers and strings, this is the value itself.
+   For arrays or vectors, the elements are converted recursively.
+   For atoms, the object is converted to a JS_Object.
+Any.to_js_object : JS_Object
+Any.to_js_object self =
     m = Meta.meta self
     case m of
         _ : Meta.Atom ->
+            type_pair = ["type", Meta.get_qualified_type_name self . split '.' . last]
             cons = m.constructor
-            fs = m.fields
-            fnames = cons.fields
-            json_fs = 0.up_to fnames.length . fold Map.empty m-> i->
-                m.insert (fnames.at i) (fs.at i . to_json)
-            with_tp = json_fs . insert "type" (Json.String cons.name)
-            Json.Object with_tp
+            java_cons = cons.value ...
```

Comment:
Why is this called `java_cons`?

Reply:
Because it is a BuiltIn Constructor not the Enso Meta Constructor.